lib/linguist/heuristics.rb

```diff
@@ -75,6 +75,11 @@ def call(data)
     end
 
     # Common heuristics
+    CRegex = Regexp.union(
+        /^(char|double|short|int|long|float|unsigned|void)*\w\s*\w\s*\(\s*void\s*\)/,
+        /^(\w|\s|\*)*restrict/,
+        /^(char|short|int|long|float|unsigned)*\s+\w+\s*\[\]\s*;/
```

Comment:
Why the asterisk after `(char|short|int|long|float|unsigned)`?

Reply:
good catch

Reply:
I think there's still the same error with the two above regular expressions, or am I mistaken?

Reply:
No, because u need a return type for any void function, like ` foo(void)` with no return type is not valid.

for the second one i think it's optional to have a word character or a `*` pointer type, since I want to match just the restrict keyword.

Reply:
> No, because u need a return type for any void function, like ` foo(void)` with no return type is not valid.

Sure, but currently, you can have several return types because of the asterisk. For example, `chard d(void)` matches your regular expression. Also, I think you should require a space after the type and probably meant to write `\w+` instead of just `\w`. Not sure why you have `\w\s*` twice after the type.

> for the second one i think it's optional to have a word character or a `*` pointer type, since I want to match just the restrict keyword.

Could you give an example of C construction you're trying to match with the second regular expression?

Reply:
> Sure, but currently, you can have several return types because of the asterisk. For example, doublechard d(void) matches your regular expression. Also, I think you should require a space after the type and probably meant to write \w+ instead of just \w. Not sure why you have \w\s* twice after the type.

Alright, will fix this.

> Could you give an example of C construction you're trying to match with the second regular expression?

Would be:

```c
void copy(void* restrict dst, void* restrict src){...}
```

The restrict keyword is always used with pointers, but most always used in function parameters.

Reply:
Maybe `\([\s\w\*]+restrict\s+` for the second expression then?